server/src/main/java/org/elasticsearch/indices/SystemIndices.java

```diff
@@ -116,7 +118,8 @@ public class SystemIndices {
      * This is the source for non-plugin system features.
      */
     private static final Map<String, Feature> SERVER_SYSTEM_FEATURE_DESCRIPTORS = Stream.of(
-        new Feature(TASKS_FEATURE_NAME, "Manages task results", List.of(TASKS_DESCRIPTOR))
+        new Feature(TASKS_FEATURE_NAME, "Manages task results", List.of(TASKS_DESCRIPTOR)),
+        new Feature(SYNONYMS_FEATURE_NAME, "Manages synonyms", List.of(SYNONYMS_DESCRIPTOR))
```

Comment:
Should we add .synonyms index only `SYNONYMS_API_FEATURE_FLAG` is enabled?

Reply:
Good catch! Adding that 👍